fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt

```diff
@@ -92,6 +98,23 @@ class SecretSettingsFragment : PreferenceFragmentCompat() {
             onPreferenceChangeListener = SharedPreferenceUpdater()
         }
 
+        lifecycleScope.launch {
+            val debugDataStore = requireContext().debugSettings
+
+            // During initial development, this will only be available in Nightly or Debug builds.
+            requirePreference<SwitchPreference>(R.string.pref_key_enable_debug_drawer).apply {
+                isVisible = Config.channel.isNightlyOrDebug
+                isChecked = debugDataStore.debugDrawerEnabled.first()
```

Comment:
nit: IMO I think moving the `first()` call directly to `debugDrawerEnabled` would keep the responsibility in a single place instead of the caller having to make the call on a `flow` as we only want the boolean value and the flow won't update after the first call? e.g.
```kotlin
suspend fun DataStore<Preferences>.debugDrawerEnabled(): Boolean = data.map { preferences ->
        preferences[DEBUG_DRAWER_ENABLED] ?: false
    }.first()
```
Not a blocker though

Reply:
We'll need the observe the flow as part of https://github.com/mozilla-mobile/firefox-android/pull/4612/commits/32fc424e1482913e97c21c478ae88062b9bed65b. 

Should we keep the `single` in that case, or maybe instead just expose a non-observable API?

Reply:
Ah okay, so just checking my understanding:
* Prefs screen needs the `Boolean` value only as this retrieves the value only when created
* The overlay needs to observe the `Flow` to update its enabled state as this is always present
?

Reply:
That's correct!